upgradeability_using_generic_eternal_storage/contracts/test/OwnedToken.sol

```diff
@@ -34,7 +34,7 @@ contract OwnedToken is EternalStorage {
    * @dev Allows the current owner to transfer control of the contract to a newOwner.
    * @param newOwner The address to transfer ownership to.
    */
-  function transferOwnership(address newOwner) public onlyTokenOwner {
+  function transferTokenOwnership(address newOwner) public onlyTokenOwner {
```

Comment:
why can't you use vanilla `Ownable` for this?

Reply:
The problem is that we need to handle two different ownerships: one for upgradeability purpose and another one for the token. This means, having two different ways of transferring ownerships. We could have used `transferOwnership` for one of them, but it wouldn't have been clear enough to use a custom function name in one of the contracts and a generic one in the other.

Bear in mind that it is not possible to use `trasnferOwnership` for both contracts given we are using the fallback function of the `Proxy` to delegate requests to the last upgraded token behavior. 

On the other hand, we could have put the `OwnedToken` behavior inside the `Token_V1` itself. But I tried to followed the inheritance approach as if we were using OZ. Although, you will see that [`Token_V3`](https://github.com/zeppelinos/labs/pull/9/files#diff-95e31b5b2498433fe07c9ac815b8bfefR11) is a pausable token that implements `Pausable` behavior itself, I didn't model a `Pausable` contract and inherited from it. We probably may want to choose between one of the two approaches, but I wanted to give an example to show both ways and discuss about them.